### ISSUE TITLE
geoserver - compile with extensions for mapstore2

### ISSUE THREAD

**Issue by fvanderbiest** [georchestra/georchestra#3081]
The goal here is to have MapStore2-linked extensions:
 * shipped in the `georchestra/geoserver` (geofence flavoured or not) docker image
 * filled in the Makefile

Below the list of GeoServer plugins/extensions used by MapStore:
- wmts-multidimensional (mandatory if you use the Timeline tool)
- Cross Layer Filtering (used by the Query builder)
- WPS (used by Attribute table, Widgets and Query builder)
- DXF (optional, only if you whant to export vectorial data in this format from the Attribute table tool) 
- Excel (optional, only if you whant to export vectorial data in this format from the Attribute table tool)
- CSS plugin (optional, used by the Style Editor)
- DDS/BIL plugin.

**Comment by landryb** [comment]
even if its not required by mapstore2 anymore, care to add authkey module ? been using it recently, and it can be useful for some usecases.... (ie provide an url to an external contractor consuming a private geoserver endpoint without giving him an ldap account)

**Comment by pmauduit** [comment]
you set 20.0.5 on this issue, but 20.1 on the geoserver 2.17 upgrade

**Comment by pmauduit** [comment]
> even if its not required by mapstore2 anymore, care to add authkey module ?

Feel free to contribute to the https://github.com/georchestra/georchestra/tree/gs-2.17.0-update branch ;-)

**Comment by fvanderbiest** [comment]
> @fvanderbiest you set 20.0.5 as milestone on this issue, but 20.1 on the geoserver 2.17 upgrade

My mistake. Sorry

**Comment by landryb** [comment]
> > even if its not required by mapstore2 anymore, care to add authkey module ?
> 
> Feel free to contribute to the https://github.com/georchestra/georchestra/tree/gs-2.17.0-update branch ;-)

well i have no idea what im doing and maven hates me, but here you go:

https://github.com/georchestra/georchestra/pull/3070/commits/3d4133c5c32888a50b5b2206a6981f7d19e19662

of course, gs-authkey jar is nowhere to be found in maven repos since repo.boundlessgeo.com is still dead and #3061 is not closed, but .. *shrug*

**Comment by landryb** [comment]
Ah, apparently its been promoted from community to extensions: https://mvnrepository.com/artifact/org.geoserver.extension/gs-authkey/2.17.0

**Comment by landryb** [comment]
```
[WARNING] The requested profile "mapstore2" could not be activated because it does not exist.
```
really, i have no idea what im doing.

**Comment by jusabatier** [comment]
Is it planned to add mapstore2 dependencies to the WAR generated at https://packages.georchestra.org/bot/wars/20.0.x/ ?

**Comment by pmauduit** [comment]
> Is it planned to add mapstore2 dependencies to the WAR generated at https://packages.georchestra.org/bot/wars/20.0.x/ ?


once the PR is merged it will be available, but I think for version 20.1, not 20.0.x

**Comment by pmauduit** [comment]
github-actions & makefile have been modified in the gs-2.17 branch, so this issue should be addressed when #3070 will be merged.

**Comment by pmauduit** [comment]
gs 2.17.2 branch just merged, closing.
